Sticky case tile in web apps

orangejenny opened the pull request:
## Product Description

https://dimagi-dev.atlassian.net/browse/USH-3980

Make persistent case tiles stick to the top of the screen. This will affect all persistent case tiles, whether they're in a case list or a form.

Also makes a few minor styling adjustments. Review by commit.

<img width="1265" alt="Screen Shot 2023-11-29 at 8 23 37 PM" src="https://github.com/dimagi/commcare-hq/assets/1486591/22587341-735c-4bfd-b6dc-cd9ae817e575">
<img width="1273" alt="Screen Shot 2023-11-29 at 8 23 47 PM" src="https://github.com/dimagi/commcare-hq/assets/1486591/364b7787-c9da-4cf4-8483-11b5389b5e73">

## Feature Flag
Case tiles

## Safety Assurance

### Safety story
This is all styling changes, but I'm requesting QA to make sure none of the "cleanup" has unwanted side effects.

### Automated test coverage

doubt it

### QA Plan

https://dimagi-dev.atlassian.net/browse/QA-5877

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

orangejenny committed:
Moved the container that contains form entry into the overall .content-container

This is a semantic change: content belongs in the content container, not after it.

orangejenny committed:
Removed extra container class

Now that #webforms is inside .content-container, which is also a .container,
it's getting double padding. Fix that.

orangejenny committed:
Removed containing .row and .col-sm-12 classes

There are enough outer containers.

orangejenny committed:
Removed top margin from form entry container

Unnecessary and causes inconsistency with other screens

orangejenny committed:
Added sticky styling to persistent case tile

orangejenny committed:
Added padding specific to app preview

This may be why #webforms was outside of .form-container: on app
preview, there's no .form-container. Deal with this with a style
in an app preview file.

orangejenny committed:
Removed uppercase transformation for required field validations on app preview

Shhhhh

orangejenny committed:
Updated margins on .form-actions

This is another side effect of removing the double .container
Also nest some related styles together.

orangejenny committed:
Updated B5 diffs

orangejenny committed:
Merge branch 'master' into jls/web-apps-sticky-tile

orangejenny committed:
Ran build_bootstrap5_diffs

nospame commented on corehq/apps/hqwebapp/static/cloudcare/less/formplayer-webapp/form.less:
```diff
@@ -134,6 +134,18 @@
 
+.case-tile-container {
+    position: sticky;
+    position: -webkit-sticky;
```
Is this still necessary? It looks like Safari supports regular `position: sticky` since late 2019.

nospame reviewed (commented):
> This will affect all persistent case tiles, whether they're in a case list or a form

Does this interact at all with the sticky map (any screen size) or map+header (on small screens)?

orangejenny committed:
Merge branch 'master' into jls/web-apps-sticky-tile

orangejenny committed:
Ran build_bootstrap5_diffs

orangejenny committed:
Removed webkit-specific style

orangejenny committed:
Moved styling to inner div

Otherwise, a border & shadow appear even when there's no tile.

orangejenny committed:
Merge branch 'jls/web-apps-sticky-tile' of github.com:dimagi/commcare-hq into jt/anchored-submit-bar-visual-fix+jls/web-apps-sticky-tile

orangejenny committed:
Merge branch 'master' into jls/web-apps-sticky-tile

orangejenny committed:
Updated new styles to not apply to printing

orangejenny committed:
Added specific z-index to resolve overlap with navbar dropdown menu

orangejenny committed:
Merge branch 'master' into jls/web-apps-sticky-tile

biyeun commented on corehq/apps/hqwebapp/static/hqwebapp/less/_hq/includes/variables.less:
```diff
@@ -97,6 +97,7 @@
 @checkbox-default-color: @call-to-action-mid;
 
 @breadcrumb-separator: '>';
+@breadcrumb-height-cloudcare: 46px;
```
yes, correct. you can add that in

orangejenny committed:
Removed auto-add of jQuery UI classes

orangejenny committed:
Removed jquery UI classes from base_cloudcare_template.html

This is just mimicking web apps HTML.

orangejenny committed:
Removed custom styles from builds/edit_menu.html

There isn't a .container on the page anymore, and the table doesn't need a custom line height.

orangejenny committed:
Updated styles for .notifications-container

orangejenny committed:
Removed border removal

I suspect this was added to remove the border added by .ui-widget-content

orangejenny committed:
Removed a.btn-primary rule

This was an override for a .ui-widget-content rule that set link text to dark grey.

orangejenny committed:
Ran build_bootstrap5_diffs

orangejenny committed:
Linted

orangejenny committed:
Ran build_bootstrap5_diffs

orangejenny commented:
Requesting review of recent commits.

Fixed https://dimagi-dev.atlassian.net/browse/QA-5983 with https://github.com/dimagi/commcare-hq/pull/33819/commits/e9d5bc9d43c2102c2e75ec1bc626128b374ee1bd, which removed a 10+-year-old line that automatically adds `ui-widget` and `ui-widget-container` classes to all `container` classes. `ui-widget` and `ui-widget-container` are jQuery UI classes, so I figure that dates back to a time when jQuery UI was used much more extensively. I've convinced myself this isn't that big a deal grepping the codebase for `container` and for `ui-widget`. Hardly any pages even use `container` outside of web apps, and they're mostly simple pages that don't use jQuery UI widgets (domain_request.html, activate_transfer_domain.html, etc.).

FYI @biyeun

biyeun commented:
@orangejenny noting that we chatted about this offline and that my recommendation is to put this change on staging and see if it causes any issues. if not it's g2g from my perspective

orangejenny commented:
@biyeun yep, it's been on staging for a few days without issue and I've done some light smoke testing on pages using jQuery UI

stephherbers reviewed (approved):
thank you for the detailed commit messages
